code-orange/VisitedLocationPanel.swift

```diff
@@ -0,0 +1,233 @@
+  }()
+  
+  private lazy var titleStackView: UIStackView = {
+    let stackView = UIStackView()
+    stackView.axis = .vertical
+    stackView.spacing = 2
+    stackView.alignment = .center
+    stackView.addArrangedSubview(pagesLabel)
+    stackView.addArrangedSubview(addressLabel)
+    stackView.addArrangedSubview(timeframeLabel)
+    return stackView
+  }()
+
+  private lazy var titleAndButtonsStackViewContainer: UIView = {
+    let stackView = UIStackView()
+    stackView.backgroundColor = .nxGrey10
+    stackView.axis = .horizontal
+    stackView.spacing = 2
+    stackView.alignment = .center
+    stackView.isLayoutMarginsRelativeArrangement = true
+    stackView.layoutMargins = UIEdgeInsets(top: 12, left: 16, bottom: 16, right: 16)
+    stackView.addArrangedSubview(leftPageButton)
+    stackView.addArrangedSubview(titleStackView)
+    stackView.addArrangedSubview(rightPageButton)
+    let view = UIView()
+    view.backgroundColor = .nxGrey10
+    view.addSubview(stackView)
+    stackView.translatesAutoresizingMaskIntoConstraints = false
+    stackView.fillSuperview()
+    return view
+  }()
+
+  private lazy var actionsStackView: UIStackView = {
+    let stackView = UIStackView()
+    stackView.axis = .horizontal
+    stackView.distribution = .fillEqually
+    stackView.spacing = 24
+    stackView.alignment = .center
+    stackView.isLayoutMarginsRelativeArrangement = true
+    stackView.layoutMargins = UIEdgeInsets(top: 12, left: 24, bottom: 24, right: 24)
+    stackView.addArrangedSubview(healthAdministrationWebsiteButton)
+    stackView.addArrangedSubview(callEmergencyServiceButton)
+    return stackView
+  }()
+
+  private lazy var leftPageButton: UIButton = {
+    let button = UIButton()
+    button.setImage(UIImage(named: "left"), for: .normal)
+    button.addTarget(self, action: #selector(previousLocationTapped), for: .touchUpInside)
+    return button
+  }()
+
+  private lazy var rightPageButton: UIButton = {
+    let button = UIButton()
+    button.setImage(UIImage(named: "right"), for: .normal)
+    button.addTarget(self, action: #selector(nextLocationPickedTapped), for: .touchUpInside)
+    return button
+  }()
+
+  private lazy var pagesLabel: UILabel = {
+    let label = UILabel()
+    label.font = UIFont.systemFont(ofSize: 16)
+    return label
+  }()
+
+  private lazy var addressLabel: UILabel = {
+    let label = UILabel()
+    label.font = UIFont.boldSystemFont(ofSize: 16)
+    return label
+  }()
+  
+  private lazy var timeFormatter: DateFormatter = {
+    let dateFormater = DateFormatter()
+    dateFormater.dateFormat = "HH:MM"
+    return dateFormater
+  }()
+
+  private lazy var timeframeLabel: UILabel = {
+    let label = UILabel()
+    label.font = UIFont.systemFont(ofSize: 14)
+    return label
+  }()
+
+  private lazy var topLabelStack: UIStackView = {
+    let paragraphStyle = NSMutableParagraphStyle()
+    paragraphStyle.alignment = .justified
+    paragraphStyle.baseWritingDirection = .rightToLeft
```

Comment:
would that work on english? why not natural alignment?

Reply:
Natural didn't work, that wouldn't work with English. But also there are other localization issues that I thought we would solve when that flow needs support